tests/integ/test_airflow_config.py

```diff
@@ -0,0 +1,651 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import gzip
+import os
+import pickle
+import sys
+
+import numpy as np
+
+from sagemaker import (
+    KMeans,
+    FactorizationMachines,
+    IPInsights,
+    KNN,
+    LDA,
+    LinearLearner,
+    NTM,
+    PCA,
+    RandomCutForest,
+)
+from sagemaker.amazon.amazon_estimator import registry
+from sagemaker.amazon.common import read_records
+from sagemaker.chainer import Chainer
+from sagemaker.estimator import Estimator
+from sagemaker.mxnet import MXNet
+from sagemaker.pytorch.estimator import PyTorch
+from sagemaker.sklearn import SKLearn
+from sagemaker.tensorflow import TensorFlow
+from sagemaker.xgboost.defaults import XGBOOST_LATEST_VERSION
+from sagemaker.workflow import airflow as sm_airflow
+from sagemaker.utils import sagemaker_timestamp
+
+import airflow
+from airflow import DAG
+from airflow.contrib.operators.sagemaker_training_operator import SageMakerTrainingOperator
+from airflow.contrib.operators.sagemaker_transform_operator import SageMakerTransformOperator
+
+from sagemaker.xgboost import XGBoost
+from tests.integ import DATA_DIR, PYTHON_VERSION
+from tests.integ.record_set import prepare_record_set_from_local_files
+from tests.integ.timeout import timeout
+
+from six.moves.urllib.parse import urlparse
+
+PYTORCH_MNIST_DIR = os.path.join(DATA_DIR, "pytorch_mnist")
+PYTORCH_MNIST_SCRIPT = os.path.join(PYTORCH_MNIST_DIR, "mnist.py")
+AIRFLOW_CONFIG_TIMEOUT_IN_SECONDS = 10
+
+RESOURCE_PATH = os.path.join(os.path.dirname(__file__), "..", "data")
+TF_MNIST_RESOURCE_PATH = os.path.join(RESOURCE_PATH, "tensorflow_mnist")
+SCRIPT = os.path.join(TF_MNIST_RESOURCE_PATH, "mnist.py")
+ROLE = "SageMakerRole"
+SINGLE_INSTANCE_COUNT = 1
+
+
```

Comment:
is there any value in marking some of these for canaries?

Reply:
Absolutely. This whole module runs in <5 seconds and doesn't utilize any ml instance, so has no flakiness. I'll mark all of these tests for canaries in my next commit.